scripts/run_asan_suite.sh

```
@@ -0,0 +1,23 @@
+cd "$ROOT_DIR"
+
+HARDENING_TOOLCHAIN="${HARDENING_TOOLCHAIN:-nightly-2025-07-14}"
+SANITIZER_TARGET="${SANITIZER_TARGET:-$(rustc +${HARDENING_TOOLCHAIN} -vV | awk '/^host: / { print $2 }')}"
+
+source "$ROOT_DIR/scripts/hardening_test_names.sh"
+
+export ASAN_OPTIONS="${ASAN_OPTIONS:-detect_leaks=0}"
+export RUSTFLAGS="${RUSTFLAGS:-} -Zsanitizer=address"
+export RUSTDOCFLAGS="${RUSTDOCFLAGS:-} -Zsanitizer=address"
+
+for test_filter in "${hardening_test_filters[@]}"; do
```

Shellcheck (0.11.0)

[warning] hardening_test_filters is referenced but not assigned.

(SC2154)